src/state/mod.rs

```diff
@@ -50,13 +51,11 @@ pub struct State {
 
 impl State {
     // Creating some of the wgpu types requires async code
-    pub async fn new(window: &Window) -> Self {
-        let (surface, adapter) = init_surface(window).await;
-
+    pub async fn new(window: Window) -> Self {
         let size = window.inner_size();
-        let config = init_config(size, &surface, &adapter);
-
+        let (surface, adapter, window) = init_surface(window).await;
```

Comment:
Why move `window` in and then return it out while it can be taken by const ref 🤨?